forecasting/src/autogluon/forecasting/utils/ml/models/mqcnn/mqcnn_model.py

```diff
@@ -0,0 +1,82 @@
+from ..abstract.abstract_model import AbstractModel
+from .default_parameters import get_default_parameters
+from gluonts.model.seq2seq._mq_dnn_estimator import MQCNNEstimator
+import os
+import time
+from core import Int, Space
+import json
+from ..abstract.model_trial import model_trial
+from tqdm import tqdm
+from core.scheduler.fifo import FIFOScheduler
+
+
+class MQCNNModel(AbstractModel):
+
+    def __init__(self, hyperparameters=None, model=None):
+        super().__init__(hyperparameters, model)
+        self.set_default_parameters()
+        self.name = "mqcnn"
+        if hyperparameters is not None:
+            self.params.update(hyperparameters)
+        self.best_configs = self.params.copy()
+
+    def set_default_parameters(self):
+        self.params = get_default_parameters()
+
+    def create_model(self):
+        # print(self.params)
+        self.model = MQCNNEstimator.from_hyperparameters(**self.params)
+
+    def fit(self, train_ds):
+        # print(self.params)
+        self.create_model()
+        self.model = self.model.train(train_ds)
+
+    def predict(self, test_ds, num_samples=100):
+        # self.model = self.model.predict(test_ds)
+        from gluonts.evaluation.backtest import make_evaluation_predictions
+        forecast_it, ts_it = make_evaluation_predictions(dataset=test_ds,
+                                                         predictor=self.model,
+                                                         num_samples=num_samples)
+        return list(tqdm(forecast_it, total=len(test_ds))), list(tqdm(ts_it, total=len(test_ds)))
+
+    def score(self, y, metric=None):
```

Comment:
why is `score()` defined in the model? Don't all models use the same scoring after their predictions are computed?

The `score` function should instead be moved to Learner.
It should look something like this:

```
def score(test_data, eval_metric=None, model=None, kwargs):
   if `eval_metric` is None:
       eval_metric = self.eval_metric  # Learner should keep track of eval_metric as an attribute

   if model is None:
        model = self.get_best_model()

   forecasts, tss = model.predict(test_data)
   ... # rest of code from your score function here...
```